cibyl/plugins/openstack/sources/jenkins_job_builder.py

```diff
@@ -0,0 +1,177 @@
+"""
+#    Copyright 2022 Red Hat
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+"""
+import logging
+import re
+import xml.etree.ElementTree as ET
+from io import StringIO
+
+from cibyl.models.attribute import AttributeDictValue
+from cibyl.models.ci.base.job import Job
+from cibyl.plugins.openstack.deployment import Deployment
+from cibyl.plugins.openstack.node import Node
+from cibyl.sources.plugins import SourceExtension
+from cibyl.sources.source import speed_index
+
+LOG = logging.getLogger(__name__)
+
+# e.g. controller:3,database:3,messaging:3,networker:2,compute:2
+TOPOLOGY = "[a-zA-Z0-9:,]+:[0-9]+"
+NODE_NAME_COUNTER = "[a-zA-Z]+:[0-9]+"
+IP_VERSION = "--network-protocol\s+ipv[4|6]"
+IP_VERSION_NUMBER = "ipv4|ipv6"
```

Comment:
nitpick: All the other sources (Jenkins, zuul, elasticsearch) return the ip version with just the number. It would be good to have consistency here, IMO

Reply:
> nitpick: All the other sources (Jenkins, zuul, elasticsearch) return the ip version with just the number. It would be good to have consistency here, IMO

Done